src/main/java/com/openshift/internal/restclient/DefaultClient.java

```diff
@@ -349,137 +293,37 @@ public  boolean supports(Class<? extends ICapability> capability) {
-			for (ModelNode node : versionNodes) {
-				try{
-					versions.add(Enum.valueOf(klass, node.asString()));
-				}catch(IllegalArgumentException e){
-					LOGGER.warn(String.format("Unsupported server version '%s' for '%s'",  node.asString(), klass.getSimpleName()));
-				}
-			}
-			return versions;
-		} catch (MalformedURLException e) {
-			LOGGER.error("Exception", e);
-			throw new OpenShiftException(e,"");
-		} catch (SocketTimeoutException e) {
-			LOGGER.error("Exception", e);
-			throw new OpenShiftException(e,"");
-		//HACK - This gets us around a server issue
-		} catch (HttpClientException e) {
-			if(e instanceof NotFoundException) {
-				throw new com.openshift.restclient.NotFoundException(e);
-			}
-			if(e.getResponseCode() != 403) {
-				throw e;
-			}
-			LOGGER.error("Unauthorized exception. Can system:anonymous get the API endpoint", e);
-			return new ArrayList<T>();
-		}
-	}
-
 	@Override
 	public URL getBaseURL() {
 		return this.baseUrl;
 	}
 
-	@SuppressWarnings("deprecation")
 	@Override
-	public void setAuthorizationStrategy(IAuthorizationStrategy strategy) {
-		this.strategy = strategy;
-		this.client.setAuthorizationStrategy(strategy);
+	public IAuthorizationContext getAuthorizationContext() {
+		return this.authContext;
 	}
 	
-	@Override
-	public IAuthorizationStrategy getAuthorizationStrategy() {
-		return this.strategy;
-	}
-
-	private OpenShiftException createOpenShiftException(String message, HttpClientException e) {
-		LOGGER.debug(message, e);
-		final String token = strategy != null ? strategy.getToken() : "";
-		if (e.getMessage() != null
-				&& e.getMessage().startsWith("{")) {
-			IStatus status = factory.create(e.getMessage());
-			if(status.getCode() == STATUS_FORBIDDEN) {
-				if(StringUtils.isNotBlank(token)) { //truly forbidden
-					return new ResourceForbiddenException(status.getMessage(), status, e);
-				}else {
-					return new com.openshift.restclient.authorization.UnauthorizedException(authClient.getAuthorizationDetails(this.baseUrl.toString()));
-				}
-			}
-			return new OpenShiftException(e, status, message);
-		} else {
-			if(e instanceof UnauthorizedException) {
-				return new com.openshift.restclient.authorization.UnauthorizedException(authClient.getAuthorizationDetails(this.baseUrl.toString()));
-			}
-			return new OpenShiftException(e, message);
-		}
-	}
-
-	@Override
-	public IUser getCurrentUser() {
-		return get(ResourceKind.USER, "~", "");
+	public void setToken(String token) {
+		this.authContext.setToken(token);
 	}
-
-	@Override
-	public IAuthorizationContext getContext(String baseURL) {
-		return this.authClient.getContext(baseURL);
-	}
-
-	@Override
-	public IAuthorizationDetails getAuthorizationDetails(String baseURL) {
-		return this.authClient.getAuthorizationDetails(baseURL);
+	
+	public String getToken() {
+		return getAuthorizationContext().getToken();
 	}
 
 	@Override
-	public void setSSLCertificateCallback(ISSLCertificateCallback callback) {
```

Comment:
We do but through the initialization of the client via the ClientBuilder.  I couldn't find a case where it needed to exist as a setter on the client.  In the JBT code we set the callback when we are creating new connections which rely on ClientBuilder